Automattic/jetpack
# Allow for options to be retrieved by the sync/object endpoint

This update allows us to expand the checksum process so that we ensure options are always in sync. It implements get_objects_by_ids and get_object_by_id for the Options module that allows for us to retrieve the current option value on demand. Pulled options are limited to the allowed Sync options and does not give any visibility to not allowed options.

#### Changes proposed in this Pull Request:

Expand sync/object endpoint to include retrieval of options.

#### Does this pull request change what data or activity we track or use?


No options are already retrievable via a Full Sync this simply exposes a new endpoint to request the data.

#### Testing instructions:






This PR is building off of https://github.com/Automattic/jetpack/pull/19959 but can be tested using either the json-api or the v4 rest-api.

1) Access shell on your WP.com sandbox
2) $jsm = new Jetpack_Sync_Manager( get_current_blog_id() );
3) $jsm->get_sync_objects( 'options', 'option', array( 'all' ) );
4) Verify that options are retrieved
5) $jsm->get_sync_objects( 'options', 'option', array( 'blue-bird' ) );
6) Verify that empty option array is returned
7) Specify your own option names in the id only allowed ids will be returned.
8) Apply D62139 and re-run tests to verify that the v4 REST API also returns same values.
